### DISCUSSION TITLE
Prisma should add more advance frature like query method , schema write in multiple .prisma file

### DISCUSSION THREAD

**Discussion by mdarifulislamroni521** [prisma/prisma#24002] [Q&A]
### Question

last 1 year I used Prisma but I think Prisma should update with some new features like model query method for automatic process data in query like Python class method. and it's a very big problem if we use Prisma in the biggest project that write all models in 1 schema file. it's tough to read and manage 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by janpio** [comment]
We are working on supporting multiple Prisma schema files: https://github.com/prisma/prisma/issues/2377

Can you elaborate on what you mean by "query method" and "model query method for automatic process data in query like Python class method"?

**Reply by mdarifulislamroni521** [reply]
Thank you. I committed on #2377 those that I expect to Prisma 
https://github.com/prisma/prisma/issues/2377#issuecomment-2082331403

**Reply by janpio** [reply]
The first half of your comment there is offtopic, which is why you got all those 👎 votes. Many, many people in that issue got a notification about your response, so I think it is understandable why they are confused why you talk about an unrelated problem there. I hid your comment to not confuse more people, let's continue the discussion here.

I am repeating what you wrote here about the method:

> i suggest that Prisma add a query method for example we run a query const users=await prisma.users.findMany({}).apply_method(). apply_method should apply every single user object and obviously, apply_method could process and change the field value as a response like in the python class method can change self-value using self.field_name="updated_value".

**Reply by janpio** [reply]
> apply_method should apply every single user object and obviously, apply_method could process and change the field value as a response like in the python class method can change self-value using self.field_name="updated_value".

Where does `apply_method` comes from? Is it by Prisma, or defined by you, the user?
What is the result of calling this method?

**Reply by mdarifulislamroni521** [reply]
apply_method will be a custom-defined function that will execute before the response and it will return an updated object for each object. for example prima finds an object like {userId:_id,name:"user_name",age:15} then this object will pass to apply_method(can be any name) and apply_method will return like {userId:_id,name:"user_name",age:28} this similar

**Reply by janpio** [reply]
I think something similar could probably be achieved with https://www.prisma.io/docs/orm/prisma-client/client-extensions.
Do you have an example for some other tools that offers this, and how they implemented it?

**Reply by mdarifulislamroni521** [reply]
yeah i was already tested it but its work only single query not work with many query and automatically

**Reply by mdarifulislamroni521** [reply]
I have another issue when a new property is added to the model with default it should create a property on the database automatically when run db push(if not exist) with a default value else it cann't find when adding a condition for this property

**Reply by raphaelumoh** [reply]
@mdarifulislamroni521 We now have support for multiple schema files. https://www.prisma.io/blog/organize-your-prisma-schema-with-multi-file-support

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
